### DISCUSSION TITLE
httpUpdate: Does it work over cellular network?

### DISCUSSION THREAD

**Discussion by tipo1000** [espressif/arduino-esp32#8162] [Q&A]
I have httpUpdate (FOTA) working over Wi-Fi, but would it work over cellular network also?

I have SIM7000x modem which supports pretty much all network protocols there is.

What needs to change here (from example):

    WiFiClient client;
    t_httpUpdate_return ret = httpUpdate.update(client, "http://server/file.bin");

Thanks,
Tipo

**Comment by mrengineer7777** [comment]
How are you connecting the ESP32 to the cell network?  Over a WiFi hotspot?  Then it will work the same as a regular WiFi network.

**Reply by tipo1000** [reply]
ESP32 is connected to cell network with SIM7000E module (Cat-M/NB-IoT/GSM).

**Reply by mrengineer7777** [reply]
I see https://www.dfrobot.com/product-1732.html
That module likely interfaces it to the ESP32 with I2C or UART communications channel.  It is possible to update the ESP32 over serial, but I haven't tried it.  You'll also have to research how to do HTTP calls on that modem.
